src/sentry/sentry_apps/models/sentry_app_installation.py

```diff
@@ -241,6 +242,6 @@ def prepare_ui_component(
             component=component, install=installation, project_slug=project_slug, values=values
         ).run()
         return component
-    except APIError:
+    except (APIError, ValidationError):
         # TODO(nisanthan): For now, skip showing the UI Component if the API requests fail
```

Comment:
Lol! This TODO seems very old and important. Basically what to do here is to return the error and surface it to the user. Future PRs though.